src/blocks/net.rs

```diff
@@ -854,41 +867,52 @@ impl Block for Net {
 
         self.update_tx_rx()?;
 
+        let empty_string = "".to_string();
+        let s_up = format!(
+            "{} {}",
+            self.config
+                .icons
+                .get("net_up")
+                .cloned()
+                .unwrap_or_else(|| "".to_string()),
+            self.output_tx.as_ref().unwrap_or(&empty_string)
+        );
+        let s_dn = format!(
+            "{} {}",
+            self.config
+                .icons
+                .get("net_down")
+                .cloned()
+                .unwrap_or_else(|| "".to_string()),
+            self.output_rx.as_ref().unwrap_or(&empty_string)
+        );
+
+        let values = map!(
+            "{ssid}" => self.ssid.as_ref().unwrap_or(&empty_string),
+            "{signal_strength}" => self.signal_strength.as_ref().unwrap_or(&empty_string),
+            "{signal_strength_bar}" => self.signal_strength_bar.as_ref().unwrap_or(&empty_string),
+            "{bitrate}" =>  self.bitrate.as_ref().unwrap_or(&empty_string),
+            "{ip}" =>  self.ip_addr.as_ref().unwrap_or(&empty_string),
+            "{ipv6}" =>  self.ipv6_addr.as_ref().unwrap_or(&empty_string),
+            "{speed_up}" =>  &s_up,
+            "{speed_down}" => &s_dn,
+            "{graph_up}" =>  self.graph_tx.as_ref().unwrap_or(&empty_string),
+            "{graph_down}" =>  self.graph_rx.as_ref().unwrap_or(&empty_string)
+        );
+
+        self.output
+            .set_text(if let Ok(s) = self.format.render_static_str(&values) {
+                s
+            } else {
+                "[invalid connection format string]".to_string()
```

Comment:
I believe we usually return a BlockError here. Is there a reason for deviating from that pattern?

Reply:
I was refering to https://github.com/greshake/i3status-rust/blob/master/src/blocks/networkmanager.rs file while doing. After you mentioned this, I realized that similar problem exists in that file also.  Also I have update the file as per this comment.